src/utils.ts

```diff
@@ -443,6 +444,26 @@ export async function chunkPromises<T>(fns: (() => Promise<T>)[], chunkSize: num
     return results;
 }
 
+/**
+ * Retries the function until it succeeds or is interrupted. The given function must return
+ * a promise which throws/rejects on error, otherwise the retry will assume the request
+ * succeeded. The promise chain returned will contain the successful promise. The given function
+ * should always return a new promise.
+ * @param {Function} promiseFn The function to call to get a fresh promise instance. Takes an
+ * attempt count as an argument, for logging/debugging purposes.
+ * @returns {Promise<T>} The promise for the retried operation.
+ */
+export function simpleRetryOperation<T>(promiseFn: (attempt: number) => Promise<T>): Promise<T> {
+    return promiseRetry((attempt: number) => {
+        return promiseFn(attempt);
+    }, {
+        forever: true,
+        factor: 2,
+        minTimeout: 3000, // ms
+        maxTimeout: 15000, // ms
```

Comment:
pre-emptive comment: we could probably expose this as an argument, but for now we don't need to change this. It'd be trivial to do later, and likely would consider a library-agnostic way of doing it.